nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/SdkPackageUpdater.cs

```diff
@@ -53,6 +60,54 @@ public static async Task UpdateDependencyAsync(
         await SaveBuildFilesAsync(buildFiles, logger);
     }
 
+    public static async Task UpdateDependencyWithConflictResolution(
```

Comment:
This function is what enables the correct behavior; updating a transitive dependency by updating a top-level one.